server/src/apub/post.rs

```diff
@@ -87,9 +127,40 @@ impl FromApub for PostForm {
     let community_actor_id = &oprops.get_to_xsd_any_uri().unwrap().to_string();
     let community = get_or_fetch_and_upsert_remote_community(&community_actor_id, &conn)?;
 
+    let thumbnail_url = match oprops.get_image_any_image() {
+      Some(any_image) => any_image
+        .to_owned()
+        .into_concrete::<Image>()?
+        .object_props
+        .get_url_xsd_any_uri()
+        .map(|u| u.to_string()),
+      None => None,
+    };
+
+    let url = oprops.get_url_xsd_any_uri().map(|u| u.to_string());
+    let (embed_title, embed_description, embed_html) = match oprops.get_preview_base_box() {
+      Some(preview) => {
+        let preview_page = preview.to_owned().into_concrete::<Page>()?;
+        let name = preview_page
+          .object_props
+          .get_name_xsd_string()
+          .map(|n| n.to_string());
+        let summary = preview_page
+          .object_props
+          .get_summary_xsd_string()
+          .map(|s| s.to_string());
+        let content = preview_page
+          .object_props
+          .get_content_xsd_string()
+          .map(|c| c.to_string());
+        (name, summary, content)
+      }
+      None => (None, None, None),
+    };
```

Comment:
We're gonna have to filter the html, otherwise someone can easily send a <script> tag.

Reply:
I think lots of sites include scripts in their embed_html, so that it can use external players and whatnot... but on the front end I don't load that html unless they click the expand button.

Reply:
Problem is that a malicious instance could change the html, and use it to distribute malware. So that is something we definitely need to prevent.